example.py

```diff
@@ -198,6 +209,221 @@ def print_rpc_device(device: RpcDevice) -> None:
+        short_mac, values["device_short_mac"]
+    )
+    settings["mqtt"]["id"] = settings["mqtt"]["id"].replace(
+        short_mac.lower(), values["device_short_mac"].lower()
+    )
+
+    settings["name"] = values["device_name"]
+    # Some devices use MAC and short MAC (uppercase/lowercase)
+    settings["wifi_ap"]["ssid"] = settings["wifi_ap"]["ssid"].replace(
+        real_mac, values["device_mac"]
+    )
+    settings["wifi_ap"]["ssid"] = settings["wifi_ap"]["ssid"].replace(
+        short_mac, values["device_short_mac"]
+    )
+    settings["wifi_sta"]["ssid"] = values["wifi"]
+
+    # Updata dataset
+    data.update({"settings": settings})
+    data.update({"shelly": shelly})
+    data.update({"status": status})
+
+    return data
+
+
+def _normalize_rpc_data(data: dict[str, Any]) -> dict[str, Any]:
+    """Normalize data for RPC devices."""
+    config = data["config"]
+    status = data["status"]
+    shelly = data["shelly"]
+
+    real_mac: str = status["sys"]["mac"]
+
+    values = _replacement_values()
+
+    # Config endpoint
+    config["sys"]["device"]["name"] = values["device_name"]
+    config["sys"]["device"]["mac"] = values["device_mac"]
+
+    # Some devices use MAC uppercase, others lowercase
+    config["mqtt"]["client_id"] = config["mqtt"]["client_id"].replace(
+        real_mac, values["device_mac"]
+    )
+    config["mqtt"]["client_id"] = config["mqtt"]["client_id"].replace(
+        real_mac.lower(), values["device_mac"].lower()
+    )
+    config["mqtt"]["server"] = values["mqtt_server"]
+
+    if config["mqtt"].get("topic_prefix"):
+        # Some devices use MAC uppercase, others lowercase
+        config["mqtt"]["topic_prefix"] = config["mqtt"]["topic_prefix"].replace(
+            real_mac, values["device_mac"]
+        )
+        config["mqtt"]["topic_prefix"] = config["mqtt"]["topic_prefix"].replace(
+            real_mac.lower(), values["device_mac"].lower()
+        )
+
+    if config["sys"].get("sntp"):
+        config["sys"]["sntp"]["server"] = values["sntp_server"]
+
+    for id_ in range(5):
+        if config.get(f"thermostat:{id_}"):
+            config[f"thermostat:{id_}"]["sensor"] = config[f"thermostat:{id_}"][
+                "sensor"
+            ].replace(real_mac.lower(), values["device_mac"].lower())
+            config[f"thermostat:{id_}"]["actuator"] = config[f"thermostat:{id_}"][
+                "actuator"
+            ].replace(real_mac.lower(), values["device_mac"].lower())
+
+        if config.get(f"switch:{id_}"):
+            config[f"switch:{id_}"]["name"] = f"{values['switch_name']} {id_}"
+
+        if config.get(f"input:{id_}"):
+            config[f"input:{id_}"]["name"] = f"{values['input_name']} {id_}"
+
+        if config.get(f"em:{id_}"):
+            config[f"em:{id_}"]["name"] = f"{values['em_name']} {id_}"
+
+        if config.get(f"script:{id_}"):
+            config[f"script:{id_}"]["name"] = f"{values['script_name']} {id_}"
```

Comment:
```suggestion
    config_prefixes = ("thermostat:", "switch:", "input:", "em:", "script:")
    for key in config:
        if key.startswith(config_prefixes):
            key_name, id_ = key.split(":")
            config[key]["name"] = f"{key_name} {id_}"
```

I think you can simplify this as the above but make `config_prefixes` a constant

Reply:
Unfortunately not for all. `thermostat` has `sensor` and `actuator`, not `name`

Reply:
So probably two loops in that case

Reply:
Done